### PR TITLE
TF-2271 Jmap endpoint auto discovery

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2271

## Dependent

- https://github.com/dietfriends/dns_client/pull/9

### 1. Found Jmap Url

https://github.com/linagora/tmail-flutter/assets/80730648/971cc284-792a-4b26-97c1-4d0ca463f10b

### 2. Not found Jmap Url

https://github.com/linagora/tmail-flutter/assets/80730648/97d86b66-8126-4fa6-98a3-2d6982e8e2b7

### 3. With Oidc

https://github.com/linagora/tmail-flutter/assets/80730648/693fb1e8-5d06-4b63-9cd4-38b3f9f40851

### CODE REVIEW

**Comment by chibenwa** [comment]
Don't ask for email address twice.

**Comment by chibenwa** [comment]
Can you do a demo video for c3po@poc-mail-avocat.fr ?

**Comment by hoangdat** [comment]
@chibenwa can we change the SRV record to the jmap with OIDC?

**Comment by dab246** [comment]
> Can you do a demo video for [c3po@poc-mail-avocat.fr](mailto:c3po@poc-mail-avocat.fr) ?

@chibenwa With the email `c3po@poc-mail-avocat.fr` we will not be able to find the SRV record. Below demo

https://github.com/linagora/tmail-flutter/assets/80730648/2c9fbb3c-e851-4d22-8008-72146f777d87

**Comment by dab246** [comment]
@hoangdat  Please review it.

File: pubspec.yaml [hunk 1]
```diff
@@ -86,6 +86,11 @@ dependencies:
       url: https://github.com/dab246/super_tag_editor.git
       ref: master
 
+  dns_client:
+    git:
+      url: https://github.com/dab246/dns_client.git
+      ref: twake-supported
```
Comment (hoangdat) [inline pubspec.yaml:92]: Please create the upstream PR, and add the `todo` in here:
// to do: we will change it when the PR in upstream repository will be merged

Comment (chibenwa) [reply pubspec.yaml:92]: And link upstream PR?

Comment (dab246) [reply pubspec.yaml:93]: Updated

**Comment by dab246** [comment]
@hoangdat  Please review again

File: lib/features/session/presentation/session_controller.dart
Comment (hoangdat) [inline lib/features/session/presentation/session_controller.dart:1]: we can delete it?

Comment (dab246) [reply lib/features/session/presentation/session_controller.dart:1]: It's unnecessary. We will process directly on the ReloadController

**Pull request merged**